src/providers/krb5/krb5_common.c

```diff
@@ -730,6 +731,16 @@ errno_t write_krb5info_file_from_fo_server(struct krb5_service *krb5_service,
     if (filter == NULL || filter(server) == false) {
         address = fo_server_address_or_name(tmp_ctx, server);
         if (address) {
+            port = fo_get_server_port(server);
+            if (port != 0) {
+                address = talloc_asprintf(tmp_ctx, "%s:%d", address, port);
```

Comment:
it does not matter much but `talloc_asprintf_append` looks more appropriate (as well as in the second case bellow).
What do you think?

Otherwise the PR looks good to me.

Reply:
I'd like to do that but if I do that I get a compilation warning:
`warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]`.
And if I tackle that by casting to `(char *) talloc_asprintf_append()` then I get another warning:
`warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]`.

Taking into account that we'll switch all warnings into errors I don't think it's a good idea to leave a warning in the code. So, do you have any idea on how to talloc_asprintf_append() without having any compilation warning?

Reply:
I see the point. Using _append may complicate the more common usecase where default port is used.

Thank you for the explanation

ACK

Reply:
> Taking into account that we'll switch all warnings into errors

Depends on definition of "all", but in general I'm not sure this is a really good idea.

From my point of view, it would be much better to introduce a policy to ensure new patches doesn't introduce new warnings, but I'm not convinced "zero warnings code base" goal would pay off.

Reply:
It is feasible: you do a run with and without patches and compute a diff.
Covscan is already doing this for us.